src/depackers/s404_dec.c

```diff
@@ -389,6 +389,18 @@ static int decrunch_s404(FILE *in, FILE *out, long inlen)
     goto error;
   }
 
+  /**
+   * Best case ratio of S404 sliding window:
+   *
+   *  2-3:  9b + (>=1b) -> 2-3B  ->  24:10
+   *  4-7:  9b + (>=3b) -> 4-7B  ->  56:12
+   *  8:22: 9b + (>=6b) -> 8-22B -> 176:15
+   *  23+:  9b + 3b + 8b * floor((n-23)/255) + 7b + (>=0b) -> n B -> ~255:1
+   */
+  if (pLen < (oLen / 255)) {
+    goto error;
+  }
```

Comment:
Looks sane. Tested on valid files?

Reply:
Only on whatever is in test-dev/. Do you have a link to other files I could test this on?

Reply:
OK, tested with four files I had around, loads OK.